cluster-autoscaler/core/scaledown/actuation/actuator.go

```diff
@@ -96,47 +96,52 @@ func (a *Actuator) ClearResultsNotNewerThan(t time.Time) {
 	a.nodeDeletionTracker.ClearResultsNotNewerThan(t)
 }
 
+// ScaleDownStatus returns current scale down status.
+func (a *Actuator) ScaleDownStatus() *status.ScaleDownStatus {
```

Comment:
The name and description for the method suggest that a complete status is returned, where in reality it only sets 2 specific fields - IMO it's misleading.

It could make sense as something like `BaseScaleDownStatus`, but then it'd still be inconsistent with how other fields are set. `UnremovableNodesInfo` is set in `RunOnce` using data obtained from `Planner`, `ScaledDownNodes` are set in `RunOnce` using data obtained from `Actuator`, `Result` is set in `RunOnce` based on various things. I see no reason why `NodeDeleteResults` and `NodeDeleteResultsAsOf` would be set in a different way (and if they are, I start wondering why it's different for them).

I'd definitely set these 2 fields in `RunOnce` like the others. The necessary data is in `NodeDeletionTracker`, which `RunOnce` currently doesn't have access to. We can either expose the data from `Actuator` by delegating to the tracker, or plumb `NodeDeletionTracker` to `StaticAutoscaler` and call it directly. The former approach probably makes more sense, as `Actuator` already exposes a method to clear the results.

Reply:
Makes sense, will move setting part back to `RunOnce` and will add `DeletionResults` as a delegate method of actuator